firebase-firestore/src/main/java/com/google/firebase/firestore/remote/AndroidConnectivityMonitor.java

```diff
@@ -78,34 +86,72 @@ private void configureNetworkMonitoring() {
       final DefaultNetworkCallback defaultNetworkCallback = new DefaultNetworkCallback();
       connectivityManager.registerDefaultNetworkCallback(defaultNetworkCallback);
       unregisterRunnable =
-          new Runnable() {
-            @Override
-            public void run() {
-              connectivityManager.unregisterNetworkCallback(defaultNetworkCallback);
-            }
-          };
+          () -> connectivityManager.unregisterNetworkCallback(defaultNetworkCallback);
     } else {
       NetworkReceiver networkReceiver = new NetworkReceiver();
       @SuppressWarnings("deprecation")
       IntentFilter networkIntentFilter = new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION);
       context.registerReceiver(networkReceiver, networkIntentFilter);
-      unregisterRunnable =
-          new Runnable() {
-            @Override
-            public void run() {
-              context.unregisterReceiver(networkReceiver);
-            }
-          };
+      unregisterRunnable = () -> context.unregisterReceiver(networkReceiver);
     }
   }
 
   private void configureBackgroundStateListener() {
-    BackgroundDetector.getInstance().addListener(this);
+    Application applicationContext = (Application) context.getApplicationContext();
```

Comment:
nit: `applicationContext` can be renamed to `application` since it is being casted to an instance of `Application`.